src/data-providers/data-helpers/fetchDescendants.ts

```diff
@@ -209,7 +210,21 @@ async function* fetchDescendants(
           }
         } else {
           thoughtIdQueue.add(childrenIds)
-          return thought
+          return {
+            ...thought,
+            // Always mark parents as pending.
+            // Mark parents as pending ONLY if at least one direct child is not yet loaded.
+            // This prevents propagating the pending state up the entire ancestor chain.
+            // updateThoughts will still clear pending once the missing direct children are loaded.
+            pending:
+              !isRoot([thought.id]) &&
+              hasChildren &&
```

Comment:
Is `hasChildren` superfluous? It's truthy if and only if `childrenIds.some(...)` is truthy.

Reply:
I agree. This was to make the intent obvious (more readable). I'll remove.

Reply:
Removed.